OsmAnd/src/net/osmand/plus/auto/TracksScreen.kt

```diff
@@ -0,0 +1,139 @@
+import net.osmand.plus.track.helpers.GPXDatabase.GpxDataItem
+import net.osmand.plus.track.helpers.GpxDbHelper
+import net.osmand.search.core.ObjectType
+import net.osmand.search.core.SearchResult
+import net.osmand.util.Algorithms
+
+class TracksScreen(
+    carContext: CarContext,
+    private val settingsAction: Action,
+    private val surfaceRenderer: SurfaceRenderer,
+    private val trackTab: TrackTab
+) : Screen(carContext) {
+    val gpxDbHelper: GpxDbHelper
+    var trackInfoLoadingCount = 0
+    var isTrackInfoLoaded = false
+
+    init {
+        gpxDbHelper = app.gpxDbHelper
+    }
+
+    override fun onGetTemplate(): Template {
+        val templateBuilder = PlaceListNavigationTemplate.Builder()
+        val title = if (trackTab.type == TrackTabType.ALL) {
+            app.getString(R.string.sort_last_modified)
+        } else {
+            trackTab.getName(app, false)
+        }
+        if (!isTrackInfoLoaded) {
+            prepareTrackItems()
+        }
+        if (isTrackInfoLoaded) {
+            templateBuilder.setLoading(false)
+            setupTracks(templateBuilder)
+        } else {
+            templateBuilder.setLoading(true)
+        }
+
+        return templateBuilder
+            .setTitle(title)
+            .setActionStrip(ActionStrip.Builder().addAction(settingsAction).build())
+            .setHeaderAction(Action.BACK)
+            .build()
+    }
+
+    private fun prepareTrackItems() {
+        for (track in trackTab.trackItems) {
+            track.file?.let { file ->
+                val item = gpxDbHelper.getItem(file) { updateTrack(track, it) }
+                if (item != null) {
+                    track.dataItem = item
+                } else {
+                    trackInfoLoadingCount++
+                }
+            }
+        }
+        isTrackInfoLoaded = trackInfoLoadingCount == 0
+    }
+
+    private fun updateTrack(trackItem: TrackItem, dataItem: GpxDataItem?) {
+        trackItem.dataItem = dataItem
+        trackInfoLoadingCount--
+        if (trackInfoLoadingCount == 0) {
+            isTrackInfoLoaded = true
+            invalidate()
+        }
+    }
+
+    private fun setupTracks(templateBuilder: PlaceListNavigationTemplate.Builder) {
+        val listBuilder = ItemList.Builder()
+        for (track in trackTab.trackItems) {
+            val title = track.name
+            val icon = CarIcon.Builder(
+                IconCompat.createWithResource(app, R.drawable.ic_action_polygom_dark))
+                .setTint(
+                    CarColor.createCustom(
+                        app.getColor(R.color.icon_color_default_light),
+                        app.getColor(R.color.icon_color_default_dark)))
+                .build()
+            var description = ""
+            var dist = 0f
+            track.dataItem?.let { dataItem ->
+                description = dataItem.nearestCityName ?: ""
+                dist = dataItem.analysis?.totalDistance ?: 0f
```

Comment:
I think we need distance to start point of track, similar to [track menu](https://github.com/osmandapp/OsmAnd/blob/08fe103fb20cf0cf30061662cad86ea2ec6b5746/OsmAnd/src/net/osmand/plus/track/fragments/TrackMenuFragment.java#L924) or [configure track items](https://github.com/osmandapp/OsmAnd/blob/f6c5313b8292030cde3ba6c2539bda37a82d5553/OsmAnd/src/net/osmand/plus/configmap/tracks/viewholders/TrackViewHolder.java#L251-L260), not the whole track distance